contribs/freight/src/main/java/org/matsim/contrib/freight/carrier/TimeWindow.java

```diff
@@ -62,4 +62,26 @@ public String toString() {
 		return "[start=" + Time.writeTime(start) + ", end=" + Time.writeTime(end) + "]";
 	}
 
+	@Override
+	public boolean equals(Object o){
+		if(o == this) return true;
+		if((o == null) || (o.getClass() != this.getClass())) {
+			return false;
+		}
+		TimeWindow t1 = (TimeWindow) o;
+		return (t1.getStart() == this.getStart() && t1.getEnd() == this.getEnd());
+	}
+
+	@Override
+	public int hashCode(){
+		int result = 59;
+		Long startLong  = Double.doubleToLongBits(start);
+		int startHash = (int) (startLong^(startLong>>>32));
+		result = 31 * result + startHash;
+		Long endLong  = Double.doubleToLongBits(end);
+		int endHash = (int) (startLong^(startLong>>>32));
```

Comment:
shouldn't `endHash` be: `endLong^(endLong>>>32)`?

Reply:
yes it should! thanks